packages/react-dom-bindings/src/server/ReactDOMFloatServer.js

```diff
@@ -0,0 +1,580 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import ReactDOMSharedInternals from 'shared/ReactDOMSharedInternals';
+import {
+  validatePreloadResourceDifference,
+  validateStyleResourceDifference,
+  validateStyleAndHintProps,
+  validateLinkPropsForStyleResource,
+  validateLinkPropsForPreloadResource,
+  validatePreloadArguments,
+  validatePreinitArguments,
+} from '../shared/ReactDOMResourceValidation';
+
+type Props = {[string]: mixed};
+
+type ResourceType = 'style' | 'font';
+
+type PreloadProps = {
+  rel: 'preload',
+  as: ResourceType,
+  href: string,
+  [string]: mixed,
+};
+type PreloadResource = {
+  type: 'preload',
+  as: ResourceType,
+  href: string,
+  props: PreloadProps,
+  flushed: boolean,
+};
+
+type StyleProps = {
+  rel: 'stylesheet',
+  href: string,
+  'data-rprec': string,
+  [string]: mixed,
+};
+type StyleResource = {
+  type: 'style',
+  href: string,
+  precedence: string,
+  props: StyleProps,
+
+  flushed: boolean,
+  inShell: boolean, // flushedInShell
+  hint: PreloadResource,
+};
+
+export type Resource = PreloadResource | StyleResource;
+
+export type Resources = {
+  // Request local cache
+  preloadsMap: Map<string, PreloadResource>,
+  stylesMap: Map<string, StyleResource>,
+
+  // Flushing queues for Resource dependencies
+  explicitPreloads: Set<PreloadResource>,
+  implicitPreloads: Set<PreloadResource>,
+  precedences: Map<string, Set<StyleResource>>,
+
+  // Module-global-like reference for current boundary resources
+  boundaryResources: ?BoundaryResources,
+};
+
+export function createResources(): Resources {
+  return {
+    // persistent
+    preloadsMap: new Map(), // preloadResources
+    stylesMap: new Map(), // styleResources
+
+    // cleared on flush
+    explicitPreloads: new Set(), // explicitPreloads
+    implicitPreloads: new Set(), // add bootstrap script to implicit preloads
+    precedences: new Map(),
+
+    // like a module global for currently rendering boundary
+    boundaryResources: null,
+  };
+}
+
+export type BoundaryResources = Set<StyleResource>;
+
+export function createBoundaryResources(): BoundaryResources {
+  return new Set();
+}
+
+export function mergeBoundaryResources(
+  target: BoundaryResources,
+  source: BoundaryResources,
+) {
+  source.forEach(resource => target.add(resource));
+}
+
+let currentResources: null | Resources = null;
+
+let previousDispatcher = null;
+export function prepareToRender(resources: Resources) {
```

Comment:
This and cleanupAfterRender are for the general ReactDOMDispatcher. Not just Float. So seems like this shouldn't be in this specialized module.

Reply:
I clarified the names but those are actually imported into formatconfig called from a identically named function. It just so happens that there are no other things we need to do in prep and cleanup so they look like a passthrough.

Long term I think I should move the module globals from float to formatconfig and have the float apis just accept resources as an argument. This will help in a few areas and will get rid of the need for an apparent float specific prepare function